### DISCUSSION TITLE
Help me, about matrices。

### DISCUSSION THREAD

**Discussion by sww804494555** [pymupdf/PyMuPDF#1932] [Looking for help] [answered]
Given the value of the matrix, how can I use it correctly when inserting an image?
Thank you very much!
![Snipaste_2022-09-20_19-57-09](https://user-images.githubusercontent.com/51779143/191254020-5cb25211-b4b9-4921-ad17-76499bb7a7b5.png)

**Comment by JorjMcKie** [comment]
Image insertion currently only supports rotation angles of 0, 90, 180, 270 degrees. Not as you indicated in your image.
However, there is a way to achieve what you want:
1. Convert your image to a (temporary) PDF
2. Insert the first (and only) page of that PDF into your target page. The respective `Page` method `show_pdf_page()` works similar to `insert_image()`  but allows arbitrary rotation angles.

Using a matrix in this process is never required and thus not supported either.

```python
imgdoc = fitz.open(<image file>)
srcbytes = imgdoc.convert_to_pdf()
src = fitz.open("pdf", srcbytes)

# 'page' is your target page:
page.show_pdf_page(rect, # rect on the page, center of image and the rect will be equal
    src, # temp image PDF
    0, # its first (and only) page number
    rotate=angle, # desired rotation angle - any float value
)
```

**Reply by sww804494555** [reply]
Can the rotation angle be calculated from the values of the matrix?

**Comment by JorjMcKie** [comment, marked as answer]
The matrix without its translation part matrix.e, matrix.f should be the product of a scale matrix `fitz.Matrix(a, d)` and a rotation matrix `fitz.Matrix(alpha)`.
I currently have no good advice other than solving the set of 4 equations behind
```python
fitz.Matrix(a, d) * fitz.Matrix(alpha) = fitz.Matrix(37, 58, 46, -29, 0, 0)
```
to determine a, d, and alpha.
A rotation matrix is defined as `fitz.Matrix(cos, sin, −sin, cos, 0, 0)`. A scale matrix has the components `fitz.Matrix(a, 0, 0, d, 0, 0)`. So you have 4 variables and 4 equations. Once you have the values for cos and sin you can compute alpha via the arcus functions.
